crates/oci/src/client.rs

```diff
@@ -111,8 +117,18 @@ impl Client {
                             "Adding new layer for asset {:?}",
                             spin_loader::to_relative(entry.path(), &source)?
                         );
-                        let layer = Self::data_layer(entry.path()).await?;
-
+                        let mut layer = Self::data_layer(entry.path()).await?;
+                        // HACK: Empty layer uploads may not be supported depending on registry implementation.
+                        // (Context: https://github.com/distribution/distribution/discussions/4029)
+                        // Here we add two 'placeholder' bytes to such layers on upload and add an annotation
+                        // that clients pulling layers can inspect, for removing the bytes when applicable.
+                        if entry.metadata()?.len() == 0 {
```

Comment:
> some implementations wouldn't accept 1-byte uploads, either

So should this be `< 2`? If so, I would probably change the annotation to something that communicates "this has padding", too.